### ISSUE TITLE
Better Solution in Gradient Descent Quiz

### ISSUE THREAD

**Issue by danielburkard** [udacity/dlnd-issue-reports#58]

**Comment by danielburkard** [comment]
In the Quiz for "Gradient Descent" in "Intro to Neural Networks" the solution is suggested as 

del_w = [ learnrate * error * nn_output * (1 - nn_output) * 1, 
          learnrate * error * nn_output * (1 - nn_output) * 2]

i think the solution would be much clearer if it was written as:

del_w = [ learnrate * error * nn_output * (1 - nn_output) * x[0], 
          learnrate * error * nn_output * (1 - nn_output) * x[1]]

or even:

del_w = learnrate * error * n_output * (1-nn_output) * x

**Comment by aknoerig** [comment]
I agree! And as far as I understood it, it would be even clearer to move the f'(h) part into the error function. Then the algorithm matches more closely the mathematical functions outlined before –– however then the tests fail, because a different result is expected for the error variable.

    # TODO: Calculate error of neural network
    error = (y - nn_output) * nn_output * (1 - nn_output)

    # TODO: Calculate change in weights
    del_w = [learnrate * error * x[0], 
             learnrate * error * x[1]]

**Comment by clapollo** [comment]
I agree. I went with 
del_w = learnrate * error * nn_output * (1 - nn_output) * x
